### PR TITLE
Fix #9353 - Kiva build error on mac with clang >= 13

### LINKED ISSUE #9353: Kiva build error on mac with clang >= 13
Issue overview
--------------

```shell
c++ --version
Apple clang version 13.1.6 (clang-1316.0.21.2)
Target: arm64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

Errors:

```
In file included from /Users/julien/Software/Others/EnergyPlus/third_party/kiva-ep/src/libkiva/Ground.cpp:9:
In file included from /Users/julien/Software/Others/EnergyPlus/third_party/kiva-ep/src/libkiva/Ground.hpp:36:
In file included from /Users/julien/Software/Others/EnergyPlus/third_party/eigen/Eigen/IterativeLinearSolvers:11:
In file included from /Users/julien/Software/Others/EnergyPlus/third_party/eigen/Eigen/SparseCore:11:
In file included from /Users/julien/Software/Others/EnergyPlus/third_party/eigen/Eigen/Core:364:
/Users/julien/Software/Others/EnergyPlus/third_party/eigen/Eigen/src/Core/util/Meta.h:312:25: error: 'result_of<Eigen::internal::scalar_conj_product_op<double> (const double &, const double &)>' is deprecated [-Werror,-Wdeprecated-declarations]
  typedef typename std::result_of<T>::type type1;
```

```
/Users/julien/Software/Others/EnergyPlus/third_party/kiva-ep/src/libkiva/Ground.cpp:330:27: error: variable 'totalQr' set but not used [-Werror,-Wunused-but-set-variable]
    double totalQc = 0.0, totalQr = 0.0;
```

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): macOS (M1, Apple Clang 13.1.6)
 - Version of EnergyPlus (if using an intermediate build, include SHA): develop, fa02272b15601487451bfb7179e06930291282dd

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - Fix #9353 
 - Kiva build error on mac with clang >= 13
      * This is a new diagnostic flag in clang 13: `-Wunused-but-set-variable`: https://clang.llvm.org/docs/DiagnosticsReference.html#wunused-but-set-parameter
      * There is no such flag in version 12 of clang:
https://releases.llvm.org/12.0.0/tools/clang/docs/DiagnosticsReference.html
 - Silence unused-but-set warnings on clang 13 (those are NEW in clang 13)

--- 

**Note: I have xcode 13.3 (bleeding edge)**. Xcode 13.3 requires a Mac running macOS Monterey 12 or later.
Xcode 13.0 does not have the same problem, hence why @nealkruis wasn't getting these, neither was CI, and why I wasn't either until I upgraded last night.

```
c++ --version
Apple clang version 13.1.6 (clang-1316.0.21.2)
Target: arm64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: cmake/CompilerFlags.cmake [hunk 1]
```diff
@@ -75,11 +75,15 @@ elseif(CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" O
...
-    target_compile_options(project_warnings INTERFACE -Wno-unused-but-set-parameter -Wno-unused-but-set-variable)
     # Suppress unused-but-set warnings until more serious ones are addressed
+    target_compile_options(project_warnings INTERFACE -Wno-unused-but-set-parameter -Wno-unused-but-set-variable)
     target_compile_options(project_warnings INTERFACE -Wno-maybe-uninitialized)
     target_compile_options(project_warnings INTERFACE -Wno-aggressive-loop-optimizations)
   elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13.0)
+      # Suppress unused-but-set warnings until more serious ones are addressed
+      target_compile_options(project_warnings INTERFACE -Wno-unused-but-set-parameter -Wno-unused-but-set-variable)
+    endif()
```
Comment (jmarrec) [inline cmake/CompilerFlags.cmake:86]: Small extract of the warnings I am seeing:

```
/Users/julien/Software/Others/EnergyPlus/src/EnergyPlus/DXCoils.cc:7715:12: warning: variable 'EvapCondAirFlow2Des' set but not used [-Wunused-but-set-variable]
    Real64 EvapCondAirFlow2Des;             // Design low speed evaporative condenser air flow for reporting
           ^
/Users/julien/Software/Others/EnergyPlus/src/EnergyPlus/DXCoils.cc:7741:10: warning: variable 'IsCoolCoilCapAutoSize' set but not used [-Wunused-but-set-variable]
    bool IsCoolCoilCapAutoSize;             // Indicator to cooling capacity autosize for reporting
         ^
/Users/julien/Software/Others/EnergyPlus/src/EnergyPlus/DXCoils.cc:7728:12: warning: variable 'MSRatedSHRUser' set but not used [-Wunused-but-set-variable]
    Real64 MSRatedSHRUser;                  // Hard-sized multispeed rated SHR for reporting
           ^
/Users/julien/Software/Others/EnergyPlus/src/EnergyPlus/DXCoils.cc:7714:12: warning: variable 'EvapCondAirFlowUser' set but not used [-Wunused-but-set-variable]
    Real64 EvapCondAirFlowUser;             // Hard-sized evaporative condenser air flow for reporting
           ^
/Users/julien/Software/Others/EnergyPlus/src/EnergyPlus/DXCoils.cc:10713:12: warning: variable 'SHRUnadjusted' set but not used [-Wunused-but-set-variable]
    Real64 SHRUnadjusted;        // SHR prior to latent degradation effective SHR calculation
           ^
/Users/julien/Software/Others/EnergyPlus/src/EnergyPlus/DXCoils.cc:10721:12: warning: variable 'CondInletHumRat' set but not used [-Wunused-but-set-variable]
    Real64 CondInletHumRat; // Condenser inlet humidity ratio (kg/kg). Zero for air-cooled condenser.
```

File: third_party/kiva-ep/src/libkiva/Ground.cpp [hunk 2]
```diff
@@ -327,7 +327,8 @@ void Ground::calculateSurfaceAverages() {
     }
 
     double totalQ = 0.0;
-    double totalQc = 0.0, totalQr = 0.0;
+    double totalQc = 0.0;
+    // double totalQr = 0.0;
```
Comment (jmarrec) [inline third_party/kiva-ep/src/libkiva/Ground.cpp:331]:
```shell
third_party/kiva-ep/src/libkiva/Ground.cpp:330:27: error: variable 'totalQr' set but not used [-Werror,-Wunused-but-set-variable]
    double totalQc = 0.0, totalQr = 0.0;
```

File: third_party/kiva-ep/src/libkiva/Ground.hpp [hunk 3]
```diff
@@ -26,6 +26,7 @@
 #pragma clang diagnostic push
 #if __clang_major__ >= 13
 #pragma clang diagnostic ignored "-Wdeprecated-copy"
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```
Comment (jmarrec) [inline third_party/kiva-ep/src/libkiva/Ground.hpp:29]:
```shell
In file included from /Users/julien/Software/Others/EnergyPlus/third_party/kiva-ep/src/libkiva/Ground.cpp:9:
In file included from /Users/julien/Software/Others/EnergyPlus/third_party/kiva-ep/src/libkiva/Ground.hpp:36:
In file included from /Users/julien/Software/Others/EnergyPlus/third_party/eigen/Eigen/IterativeLinearSolvers:11:
In file included from /Users/julien/Software/Others/EnergyPlus/third_party/eigen/Eigen/SparseCore:11:
In file included from /Users/julien/Software/Others/EnergyPlus/third_party/eigen/Eigen/Core:364:
/Users/julien/Software/Others/EnergyPlus/third_party/eigen/Eigen/src/Core/util/Meta.h:312:25: error: 'result_of<Eigen::internal::scalar_conj_product_op<double> (const double &, const double &)>' is deprecated [-Werror,-Wdeprecated-declarations]
  typedef typename std::result_of<T>::type type1;
```

**Review by nealkruis — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Thanks for looking it over @nealkruis , and thanks for the fix @jmarrec .  Merging this now, and I will be tagging 22.1 shortly.

**Pull request merged**